oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java

```diff
@@ -94,21 +92,35 @@ protected ACTION getUserAction(Cluster cluster, Path buildPath) throws FalconExc
         addOutputFeedsAsArgument(argList, cluster);
         addInputFeedsAsArgument(argList, cluster);
 
-        sparkAction.setJar(addUri(sparkFilePath, cluster));
-
-        setSparkLibFileToWorkflowLib(sparkFilePath, entity);
+        // In Oozie spark action, value for jar is either Java jar file path or Python file path.
+        validateSparkJarFilePath(sparkJarFilePath);
+        sparkAction.setJar(getSparkJarFileName(sparkJarFilePath));
+        setSparkLibFileToWorkflowLib(sparkJarFilePath, entity);
         propagateEntityProperties(sparkAction);
 
         OozieUtils.marshalSparkAction(action, actionJaxbElement);
         return action;
     }
 
-    private void setSparkLibFileToWorkflowLib(String sparkFile, Process entity) {
+    private void setSparkLibFileToWorkflowLib(Path sparkJarFilePath, Process entity) {
         if (StringUtils.isEmpty(entity.getWorkflow().getLib())) {
-            entity.getWorkflow().setLib(sparkFile);
+            entity.getWorkflow().setLib(sparkJarFilePath.toString());
+        } else {
+            String workflowLib = entity.getWorkflow().getLib() + "," + sparkJarFilePath.toString();
+            entity.getWorkflow().setLib(workflowLib);
         }
     }
 
+    private void validateSparkJarFilePath(Path sparkJarFilePath) throws FalconException {
+        if (!sparkJarFilePath.isAbsolute()) {
+            throw new FalconException("Spark jar file path must be absolute:"+sparkJarFilePath);
+        }
+    }
+
+    private String getSparkJarFileName(Path sparkJarFilePath) {
```

Comment:
this method is not really required only one call is there and we donot have any logic implemented in it.